discord/poll.py

```diff
@@ -0,0 +1,624 @@
+        self._message = message
+        self._state = state
+        self._expiry = expiry
+
+        try:
+            self._update_results(data['results'])
+        except KeyError:
+            pass
+
+        return self
+
+    def _to_dict(self) -> PollCreatePayload:
+        data: PollCreatePayload = {
+            'allow_multiselect': self.multiselect,
+            'question': self._question_media.to_dict(),
+            'duration': self.duration.total_seconds() / 3600,
+            'layout_type': self.layout_type.value,
+            'answers': [answer._to_dict() for answer in self.answers],
+        }
+        return data
+
+    def __str__(self) -> str:
+        return self._question_media.text
+
+    def __repr__(self) -> str:
+        return f"<Poll duration={self.duration} question=\"{self.question}\" answers={self.answers}>"
+
+    def __len__(self) -> int:
+        return len(self.answers)
+
+    @property
+    def question(self) -> str:
+        """:class:`str`: Returns this poll answer question string."""
+        return self._question_media.text
+
+    @property
+    def emoji(self) -> Optional[PartialEmoji]:
+        """Optional[:class:`PartialEmoji`]: Returns the emoji for this poll's question."""
+
+        return None  # As of now, polls questions don't support emojis
+
+    @property
+    def answers(self) -> List[PollAnswer]:
+        """List[:class:`PollAnswer`]: Returns a read-only copy of the answers"""
+        return self._answers.copy()
+
+    @property
+    def expires_at(self) -> Optional[datetime.datetime]:
+        """Optional[:class:`datetime.datetime`]: A datetime object representing the poll expiry.
+
+        .. note::
+
+            This will **always** be ``None`` for stateless polls.
+        """
+        return self._expiry
+
+    @property
+    def created_at(self) -> Optional[datetime.datetime]:
+        """:class:`datetime.datetime`: Returns the poll's creation time, or ``None`` if user-created."""
+
+        if not self._message:
+            return
+        return self._message.created_at
+
+    @property
+    def message(self) -> Optional[Message]:
+        """:class:`Message`: The message this poll is from."""
+        return self._message
+
+    @property
+    def total_votes(self) -> int:
+        """:class:`int`: Returns the sum of all the answer votes."""
+        return sum([answer.vote_count for answer in self.answers])
+
+    def is_finalized(self) -> bool:
+        """:class:`bool`: Returns whether the poll has finalized.
+
+        It always returns ``False`` for stateless polls.
+        """
+        return self._finalized
+
+    is_finalised = is_finalized
+
+    def copy(self) -> Self:
+        """Returns a stateless copy of this poll.
+
+        This is meant to be used when you want to edit a stateful poll.
+
+        Returns
+        -------
+        :class:`Poll`
+            The copy of the poll.
+        """
```

Comment:
This looks like a partial deep copy. I guess the name might be fine, but it's a bit weird for a method named `copy` to do a (partial) deep copy rather than a shallow copy.

Reply:
Yeah I thought of it when I created this method, it does a partial deep copy because, as the docstring says, it is meant to return a stateless poll copy when the actual object has a state. I don't know if I explain myself...